In [1]:
import pandas as pd

In [2]:
# Credit Card Approval
df_clean = pd.read_csv("clean_dataset.csv")
df_crx = pd.read_csv("crx.csv")

In [3]:
# Mental Health
df_mental = pd.read_csv("Mental health Depression disorder Data.csv")

C:\Users\User\AppData\Local\Temp\ipykernel_36784\1382788775.py:2: DtypeWarning: Columns (5,6) have mixed types. Specify dtype option on import or set low_memory=False.
  df_mental = pd.read_csv("Mental health Depression disorder Data.csv")


In [4]:
df_sleep = pd.read_csv("Time Americans Spend Sleeping.csv")

In [5]:
df_clean.shape


(690, 16)

In [6]:
df_clean.columns

Index(['Gender', 'Age', 'Debt', 'Married', 'BankCustomer', 'Industry',
       'Ethnicity', 'YearsEmployed', 'PriorDefault', 'Employed', 'CreditScore',
       'DriversLicense', 'Citizen', 'ZipCode', 'Income', 'Approved'],
      dtype='object')

In [7]:
display(df_clean.head())

Gender    Age   Debt  Married  BankCustomer     Industry Ethnicity  \
0       1  30.83  0.000        1             1  Industrials     White   
1       0  58.67  4.460        1             1    Materials     Black   
2       0  24.50  0.500        1             1    Materials     Black   
3       1  27.83  1.540        1             1  Industrials     White   
4       1  20.17  5.625        1             1  Industrials     White   

   YearsEmployed  PriorDefault  Employed  CreditScore  DriversLicense  \
0           1.25             1         1            1               0   
1           3.04             1         1            6               0   
2           1.50             1         0            0               0   
3           3.75             1         1            5               1   
4           1.71             1         0            0               0   

        Citizen  ZipCode  Income  Approved  
0       ByBirth      202       0         1  
1       ByBirth       43     560         1  
2       ByBirth      280     824         1  
3       ByBirth      100       3         1  
4  ByOtherMeans      120       0         1

### Dataset Description: Credit Card Approval (clean_dataset.csv)

This dataset contains structured information about individuals applying for credit cards. Each row represents a unique applicant.

It includes demographic, financial, and employment-related attributes that may influence the credit approval decision.

**Columns:**
- Gender, Age, Debt, Married, BankCustomer, Industry, Ethnicity
- YearsEmployed, PriorDefault, Employed, CreditScore, DriversLicense
- Citizen, ZipCode, Income, Approved

The dataset has **690 rows** and **16 columns**.

This dataset is commonly used to build binary classification models for predicting whether a credit card application will be approved.


### 🧾 Data Type Classification: Credit Card Approval (clean_dataset.csv)

**🔢 Quantitative columns:**
- Age  
- Debt  
- YearsEmployed  
- CreditScore  
- Income  

These columns contain numerical values representing measurable features of applicants — age, financial indicators, and employment history.

**🏷️ Qualitative columns:**
- Gender  
- Married  
- BankCustomer  
- Industry  
- Ethnicity  
- PriorDefault  
- Employed  
- DriversLicense  
- Citizen  
- ZipCode  
- Approved  

These columns represent categories or labels such as gender, employment status, citizenship, or application result. Many are binary (0/1), but still categorical in meaning.


In [8]:
df_crx.shape

(689, 16)

In [9]:
df_crx.columns

Index(['b', '30.83', '0', 'u', 'g', 'w', 'v', '1.25', 't', 't.1', '01', 'f',
       'g.1', '00202', '0.1', '+'],
      dtype='object')

In [10]:
display(df_crx.head())

b  30.83      0  u  g  w  v  1.25  t t.1  01  f g.1  00202  0.1  +
0  a  58.67  4.460  u  g  q  h  3.04  t   t   6  f   g  00043  560  +
1  a  24.50  0.500  u  g  q  h  1.50  t   f   0  f   g  00280  824  +
2  b  27.83  1.540  u  g  w  v  3.75  t   t   5  t   g  00100    3  +
3  b  20.17  5.625  u  g  w  v  1.71  t   f   0  f   s  00120    0  +
4  b  32.08  4.000  u  g  m  v  2.50  t   f   0  t   g  00360    0  +

### Dataset Description: CRX Dataset (crx.csv)

This is a raw version of a credit card approval dataset with anonymized and non-descriptive column names.

Each row represents a credit card application, but column names appear to be values from the first row or incorrectly parsed data (e.g., '30.83', '1.25', 't', '+').

Due to the structure, this dataset likely needs preprocessing:
- Renaming columns
- Removing or using the first row as headers
- Handling missing or categorical values

**Shape:** 689 rows × 16 columns

The dataset is useful for advanced preprocessing and feature engineering practice before applying classification models.


### 🧾 Data Type Classification: Credit Card Applications (crx.csv)

> ⚠️ Note: This dataset appears to be unprocessed. The column names are non-descriptive (e.g., "b", "30.83", "u"), likely due to using the first row as headers. Despite this, we can still classify data types based on their values.

**🔢 Likely Quantitative columns:**
- Columns containing numeric values such as:
  - 30.83  
  - 1.25  
  - 0.1  
  - 0  

These values represent continuous or discrete numbers, likely corresponding to age, income, debt, or credit score fields.

**🏷️ Likely Qualitative columns:**
- Columns containing categorical or symbolic values such as:
  - b  
  - u  
  - g  
  - w  
  - t  
  - f  
  - +  

These values are likely encoded categories, such as employment status, housing type, previous default, or class label.

Since the dataset lacks column descriptions, this classification is based on the data format rather than semantic meaning.


In [11]:
df_mental.shape

(108553, 11)

In [12]:
df_mental.columns

Index(['index', 'Entity', 'Code', 'Year', 'Schizophrenia (%)',
       'Bipolar disorder (%)', 'Eating disorders (%)', 'Anxiety disorders (%)',
       'Drug use disorders (%)', 'Depression (%)',
       'Alcohol use disorders (%)'],
      dtype='object')

In [13]:
display(df_mental.head())

index       Entity Code  Year Schizophrenia (%) Bipolar disorder (%)  \
0      0  Afghanistan  AFG  1990           0.16056             0.697779   
1      1  Afghanistan  AFG  1991          0.160312             0.697961   
2      2  Afghanistan  AFG  1992          0.160135             0.698107   
3      3  Afghanistan  AFG  1993          0.160037             0.698257   
4      4  Afghanistan  AFG  1994          0.160022             0.698469   

  Eating disorders (%)  Anxiety disorders (%)  Drug use disorders (%)  \
0             0.101855               4.828830                1.677082   
1             0.099313               4.829740                1.684746   
2             0.096692               4.831108                1.694334   
3             0.094336               4.830864                1.705320   
4             0.092439               4.829423                1.716069   

   Depression (%)  Alcohol use disorders (%)  
0        4.071831                   0.672404  
1        4.079531                   0.671768  
2        4.088358                   0.670644  
3        4.096190                   0.669738  
4        4.099582                   0.669260

### Dataset Description: Global Mental Health Disorders (Depression-focused)

This dataset provides global statistics on various mental health disorders, measured as a **percentage of the population**.

Each row represents a country (`Entity`) in a given `Year`.

**Columns include:**
- Entity, Code, Year
- Schizophrenia (%)
- Bipolar disorder (%)
- Eating disorders (%)
- Anxiety disorders (%)
- Drug use disorders (%)
- Depression (%)
- Alcohol use disorders (%)

**Shape:** 108,553 rows × 11 columns

This dataset is valuable for analyzing trends and prevalence of mental disorders globally across time.


### 🧠 Data Type Classification: Mental Health Data

**🔢 Quantitative columns:**
- Schizophrenia (%)  
- Bipolar disorder (%)  
- Eating disorders (%)  
- Anxiety disorders (%)  
- Drug use disorders (%)  
- Depression (%)  
- Alcohol use disorders (%)  

These are percentage-based indicators measuring the prevalence of mental health conditions across populations.

**🌍 Qualitative columns:**
- Entity  
- Code  
- Year  

These columns define the geopolitical and temporal context of each record: country, its code, and the year of observation.


In [14]:
df_sleep.shape

(945, 9)

In [15]:
df_sleep.columns

Index(['index', 'Year', 'Period', 'Avg hrs per day sleeping', 'Standard Error',
       'Type of Days', 'Age Group', 'Activity', 'Sex'],
      dtype='object')

In [16]:
display(df_sleep.head())

index  Year  Period  Avg hrs per day sleeping  Standard Error Type of Days  \
0      0  2003  Annual                      8.57           0.018     All days   
1      1  2004  Annual                      8.55           0.026     All days   
2      2  2005  Annual                      8.62           0.023     All days   
3      3  2006  Annual                      8.63           0.024     All days   
4      4  2007  Annual                      8.57           0.024     All days   

           Age Group  Activity   Sex  
0  15 years and over  Sleeping  Both  
1  15 years and over  Sleeping  Both  
2  15 years and over  Sleeping  Both  
3  15 years and over  Sleeping  Both  
4  15 years and over  Sleeping  Both

### Dataset Description: Time Use in Sleep by Americans

This dataset shows how much time Americans spend sleeping per day, segmented by **year, gender, age group, and type of day** (e.g., weekend vs weekday).

Each row represents a demographic segment for a specific year and day type.

**Columns:**
- Year, Period, Type of Days
- Avg hrs per day sleeping, Standard Error
- Age Group, Sex, Activity

**Shape:** 945 rows × 9 columns

This dataset is useful for studying behavioral sleep trends in the U.S. population and comparing sleep patterns across different groups over time.


### 😴 Data Type Classification: Time Americans Spend Sleeping

**🔢 Quantitative columns:**
- Avg hrs per day sleeping  
- Standard Error  

These values reflect the average daily sleep time and its statistical variation.

**👥 Qualitative columns:**
- Year  
- Period  
- Type of Days  
- Age Group  
- Activity  
- Sex  

These fields describe the structure of the observation: time frame, demographic group, and the kind of day (weekday vs weekend).
